### DISCUSSION TITLE
Computed / Watched properties

### DISCUSSION THREAD

**Discussion by MarcTroll** [nuxt/nuxt#18211] [Questions] [answered]
How can I work with [computed properties and watchers](https://v3.vuejs.org/guide/computed.html#basic-example) in nuxt 3?

Are there any possibilities?

**Comment by ghost** [comment]
Check this link for references: https://v3.vuejs.org/api/computed-watch-api.html#computed

You have access to the `watch()` and `computed()` methods when you are using the `<script setup>` format.

**Comment by wobsoriano** [comment, marked as answer]
Just use it like how you would with Vue 3

```html
<script setup lang="ts">
const first = ref('John')
const last = ref('Doe')

const fullName = computed(() => `${first.value} ${last.value}`)

watch(fullName, (val) => {
    console.log(`Your name is: ${val}`)
})
</script>
```

No need to import.
